polaris-react/src/components/Popover/Popover.scss

```diff
@@ -66,7 +66,7 @@ $vertical-motion-offset: -5px;
 }
 
 .Content-fullHeight {
-  max-height: none;
+  max-height: 100vh;
 }
```

Comment:
This seems a little bit scary. Can we double/triple check full height popovers to ensure there are no regressions.

Reply:
> TDIL about scroll positioning and `max-height` behavior! Thanks for this fix! 💯
> 
> Should we add a comment about how this value is used to prevent scroll positioning being reset?

Thanks for reviewing! I updated the initial commit with a comment.